private/bufpkg/bufmodule/targeting_module.go

```diff
@@ -26,22 +26,28 @@ import (
 
 type targetingModule struct {
 	Module
-	targetPaths                    []string
-	targetPathsAllowNotExistOnWalk bool
+	targetPaths              []string
+	pathsAllowNotExistOnWalk bool
+	excludePaths             []string
+	allowAllTargetPaths      bool
```

Comment:
The `allowAllTargetPaths` parameter is confusing (as well as how it's used in the implementation below. It looks like we only provide it when `ModuleWithOnlyExcludePaths` is used.

Why is it necessary? Is there anything we can do to consolidate these cases when no `--path` flag is provided since those cases _should_ be equivalent, unless I'm missing something.

Reply:
Yeah, I think we could rely on the fact that `len(targetPaths) == 0`. I was concerned about that being an expected state, but then it's basically impossible to set a `--path` flag without a value, so that should be fine.